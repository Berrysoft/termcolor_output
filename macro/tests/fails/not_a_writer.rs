use termcolor_output as tco;

fn main() {
    let not_a_writer = 0u32;
    tco::write!(not_a_writer, "test");
}
