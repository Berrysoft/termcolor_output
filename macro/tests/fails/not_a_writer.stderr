error[E0599]: the method `guard` exists for type `u32`, but its trait bounds were not satisfied
 --> $DIR/not_a_writer.rs:5:14
  |
5 |     colored!(not_a_writer, "test");
  |              ^^^^^^^^^^^^ method cannot be called on `u32` due to unsatisfied trait bounds
  |
  = note: the following trait bounds were not satisfied:
          `u32: WriteColor`
          which is required by `u32: termcolor_output::WriteColorGuard`
  = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)
